system-addon/lib/TopStoriesFeed.jsm

```diff
@@ -79,11 +83,51 @@ this.TopStoriesFeed = class TopStoriesFeed {
       this.storiesLastUpdated = Date.now();
       // This is filtered so an update function can return true to retry on the next run
       this.contentUpdateQueue = this.contentUpdateQueue.filter(update => update());
+
+      this.setCache(STORIES_CACHE_KEY, this.stories);
     } catch (error) {
       Cu.reportError(`Failed to fetch content: ${error.message}`);
     }
   }
 
+  loadCachedStories() {
+    this.stories = this.getCachedStories();
+    if (this.stories && this.stories.length > 0) {
+      this.dispatchUpdateEvent(this.storiesLastUpdated, {rows: this.stories});
+      this.storiesLastUpdated = Date.now();
```

Comment:
Similarly, this date isn't entirely accurate, and we could store the date in the cache as well, but unclear if we need to be accurate here as we'll load from network unconditionally on `INIT`. Although, arguably we could be smarter if we had an actual cached time especially for Topics as those refresh once every 3 hours, and it doesn't seem unlikely a user would restart firefox within 3 hours. @csadilek?

Reply:
Yeah, I was thinking I could just set it to 1. I just want to avoid the first fetch to broadcast if we already loaded from cache.

Reply:
True, we don't want a second broadcast as that could potentially swap out the stories while a user is looking at them. This raises a question though. We could potentially show really old stories, from the last time firefox fetched successfully, but I think that's probably OK?